src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -127,6 +130,10 @@ public abstract class EC2Cloud extends Cloud {
 
     private static final SimpleFormatter sf = new SimpleFormatter();
 
+    // if this system property is defined and its value points to a valid ssh private key on disk
+    // then this will be used instead of any configured ssh credential
+    private static final String SSH_KEY_PAIR_PRIVATE_KEY_FILE = "SSH_KEY_PAIR_PRIVATE_KEY_FILE";
```

Comment:
Conventionally, system property used in Jenkins follow a format of `SomeClass.class.getName() + ".camelCase"`. This looks like an environment variable.